daft/udf/__init__.py

```diff
@@ -26,6 +26,8 @@ def __call__(
         return_dtype: DataTypeLike | None = None,
         unnest: bool = False,
         use_process: bool | None = None,
+        max_retries: int | None = None,
+        on_error: str | None = None,
```

Comment:
We can improve the typing for this like so:
```suggestion
        on_error: Literal["raise"] | Literal["log"] | Literal["ignore"] | None = None,
```

Can also be applied to all the other places we take in `on_error`